### PR TITLE
LINQ to XML consolidation

### PR DESCRIPTION
**Pull request by mairaw**

### CODE REVIEW

**Comment by mairaw** [comment]
@dotnet/docs if possible, try to avoid global changes such as https://github.com/dotnet/docs/pull/18724 from touching the LINQ content on the C# and VB folders. Otherwise, it takes a while for me to resolve the merge conflicts since this PR deletes a lot of that content.

**Comment by mairaw** [comment]
@jameshkramer we still have to remove these entries from the C# and VB TOCs correct? 

@BillWagner @gewarren do you know if there's a tool to order the entries in the redirect file? I remember I did that once but not sure if I still have the script that I used.

**Comment by jameshkramer** [comment]
@mairaw @BillWagner @gewarren 

I have no plans to alter the TOCs for the C# and VB sections. Instead, a user is redirected to the merged article if he/she selects an article that has been merged and removed. You can try it out here: https://review.docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/?branch=linq-to-xml.

Maira, what did you have in mind for ordering the redirection entries? I have already ordered the C# and VB redirects alphabetically by target. I did this so it would be fairly easy to find all the articles that are redirected to a particular target.

Does anyone want me to rebase linq-to-xml onto the latest master? I would be happy to try, though I understand that this might be something that you should control.

**Comment by mairaw** [comment]
@jameshkramer that file is typically sorted by source_path, so it would be easier to diff if we introduced any bad changes if it had that same sorting as before. I once ran a tool to do that and I believe there are some scripts available to do that.

For TOC, I think we should remove those entries from the C# and VB TOC but I'll let @gewarren and @BillWagner to decide here.

**Comment by mairaw** [comment]
And I've already merged master into this branch, so we're good. No need to rebase.

**Comment by jameshkramer** [comment]
@billwagner @gewarren @mairaw @tfosmark @marteeleigh

I agree, the ideal would be a sorting tool so that we could sort and compare the before and after redirection files.

I could restore the prior order but the prior order for the redirection entries I added was chronological (the order in which articles were merged), not alphabetical. As for the entries that were already there before this project: I left them in place except the ones for the C# parent node articles that had already been removed. I merged these redirections with the ones that I added for this project, and they are sorted along with the others, alphabetically by target.

It is fine with me to let Genevieve and Bill decide about the C# and VB TOCs. On this issue, I would like to defend myself a bit: what I implemented is (with minor changes) what I proposed in the design document that I attached to the user story back in November. I really did my best to satisfy the needs of this project, and to avoid surprises and late changes.

Jim

**Comment by gewarren** [comment]
You can sort the redirection file using the Docs Authoring Pack: https://review.docs.microsoft.com/en-us/help/contribute/docs-authoring/sort-redirects?branch=master#summary.

**Comment by gewarren** [comment]
And yes, I think we should remove the entries from the C# and VB TOCs in favor of the Standard TOC. In general we don't have more than one TOC entry per article.

**Comment by jameshkramer** [comment]
I'd be happy to sort and compare two redirection files to more clearly show the changes I made, but at this point I'm not sure which two redirection files to compare. That is, I don't know which commits to take the files from. Maira?

@mairaw @gewarren @BillWagner

**Comment by mairaw** [comment]
Thanks @gewarren, that was easy 😉 

@jameshkramer so I think the only thing left is the removal of the TOC entries

**Comment by jameshkramer** [comment]
Here is what I understand from this: if a file has an name/href in the C# or VB TOC, that name/href should be removed from the TOC (that will include all the files under the LINQ to XML node and, I think, a few others).

Is that right? 

@mairaw @gewarren @BillWagner

**Comment by mairaw** [comment]
Well, this work was supposed to just move the entire LINQ to XML node, so I'd just remove those two nodes from the C# and VB TOCs.

**Comment by Youssef1313** [comment]
Consider https://github.com/dotnet/docs/issues/20270#issuecomment-680032603 before merging to avoid the need of double redirections later.

File: .openpublishing.redirection.json
Comment (Youssef1313) [inline .openpublishing.redirection.json:1444]:
```suggestion
        },
```

Comment (mairaw) [reply .openpublishing.redirection.json]: Thanks @Youssef1313. I took care of the issues on my last commit.

**Comment by mairaw** [comment]
I'm leaving this as ready for review. The only warnings are about some missing descriptions. Perhaps they were added to the current version of the files now. @jameshkramer not sure if you have one more cycle to spare and copy/add the missing descriptions?

**Comment by Youssef1313** [comment]
Note: At some point these will need to move to the new fundamentals section. Doing that later will result in double redirection.

Pinging @BillWagner to confirm if this LINQ section is in the plan for the fundamentals sections.

**Comment by jameshkramer** [comment]
@mairaw, yes, I'll fix these.

Jim

**Comment by mairaw** [comment]
LINQ is certainly fundamental content. If there's a better location for this @BillWagner, please let us know. I agree that it would be easier not have to move this content twice. But also merge conflicts are getting complicated, so I'd like to merge this content soon.

**Comment by BillWagner** [comment]
In time, this will end up in "fundamentals".  That said, I agree with @mairaw that this is a huge improvement, and the longer we wait the more merge conflicts we'll have with this PR.

Let's :shipit: as soon as the merge conflicts are fixed (again). I've approved it @mairaw

**Review by BillWagner — APPROVED** [review]
This is awesome, and let's merge it as soon as the conflicts are resolved..

**Pull request merged**
